### PR TITLE
Feat/configurable bundle title description favicon

### LINKED ISSUE #67: w4 bundle corrupted
For large carts that less that 64kb, bundle emit invalid HTML.

I have a cart size 16kb and 44kb of HTML, but browser was crash with `Unexpected identifier`

Cart:
https://github.com/eXponenta/wasm4-game/tree/main/build

Page:
https://exponenta.github.io/wasm4-game/build/my-super-game.html

This reproduced on 1.0.3 version and on 1.0.4 version of wasm4.

Lang is AsemblyScript.

### PR DESCRIPTION
**Pull request by FaberVitale**
### Description

Adds description, title and icon to `w4 bundle`

Other changes

fixes #67
fix: different behaviour between node versions  on unhandledRejection.
perf: improves parsing speed of bundle output. 
related PR #58 

### To Do

- [x] inline favicon icons
- [x] Add metadata info: build-date, wasm4 version
- [x] Evaluate the possibility to generate on runtime `div#content` and `div.gamepad` and their descendants.
- [ ] other?

### CODE REVIEW

File: cli/cli.js [hunk 1]
```diff
@@ -43,6 +43,10 @@ program.command("png2src <images...>")
 program.command("bundle <cart>")
     .description("Bundle a cartridge for final distribution")
     .option("--html <output>", "Bundle standalone HTML")
+    .option("--title <title>", "Html page title. Default value: 'Wasm 4 game'")
```
Comment (aduros) [inline cli/cli.js:46]:
```suggestion
    .option("--title <title>", "Html page title", "Wasm 4 game")
```

File: cli/cli.js [hunk 2]
```diff
@@ -43,6 +43,10 @@ program.command("png2src <images...>")
...
+    .option("--desc <description>", "Html page description")
```
Comment (aduros) [inline cli/cli.js:47]: Can we make this `--description`?

**Comment by aduros** [comment]
This looks really solid! Thank you for keeping the number of dependencies as low as possible :heart: 

Let me know when you're done and I'll merge.

File: cli/assets/bundle/html-page.hbs [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+      <div id="gamepad-action1"></div>
+      <div id="gamepad-action2"></div>
+    </div>
+    <script
+      id="wasm4-cart-json"
+      type="application/json"
+    >{{{html.wasmCartJson}}}</script>
+    <div id="wasm4-bundle-info">
+    {{#each buildInfo}}
+      <input type="hidden" name="{{this.name}}" value="{{this.value}}">
```
Comment (aduros) [inline cli/assets/bundle/html-page.hbs:39]: Would it be better for SEO to use `<meta>` tags?

```html
<meta name="generator" content="WASM-4 1.0.3">
<meta name="created" content="2021-09-07T23:04:21.553Z">
```

Comment (FaberVitale) [reply cli/assets/bundle/html-page.hbs:39]: Sure,
I've always added commit-id, version & build-timestamp inside rendered html  pages for debugging reasons,
but I guess it this context makes more sense to use the proper meta tags.

**Comment by FaberVitale** [comment]
@aduros 

> Let me know when you're done and I'll merge.

I think we can merge this PR if there are no objections regarding its current state.

I've looked a bit into the generation at runtime of `div#content` and `div.gamepad` but I was not super happy with the implementation I wrote.

We could look into improving that part in other PR.

**Comment by aduros** [comment]
Thanks a lot, this is great! :tada: :tada:

**Pull request merged**